Sources/SourceKitLSP/CapabilityRegistry.swift

```diff
@@ -75,48 +74,128 @@ public final class CapabilityRegistry {
     clientCapabilities.workspace?.didChangeWatchedFiles?.dynamicRegistration == true
   }
 
+  // MARK: Other capability queries
+
   public var clientHasDiagnosticsCodeDescriptionSupport: Bool {
     clientCapabilities.textDocument?.publishDiagnostics?.codeDescriptionSupport == true
   }
 
-  /// Dynamically register completion capabilities if the client supports it and
-  /// we haven't yet registered any completion capabilities for the given
-  /// languages.
-  public func registerCompletionIfNeeded(
-    options: CompletionOptions,
+  /// Since LSP 3.17.0, diagnostics can be reported through pull-based requests in addition to the existing push-based
+  /// publish notifications.
+  ///
+  /// The `DiagnosticOptions` were added at the same time as the pull diagnostics request and allow specification of
+  /// options for the pull diagnostics request. If the client doesn't reject this dynamic capability registration,
+  /// it supports the pull diagnostics request.
+  public func clientSupportsPullDiagnostics(for language: Language) -> Bool {
```

Comment:
Can a client support pull based diagnostics but *not* dynamic registration? Or is pull based on available through dynamic registration?